File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        remainedStudent=len(students)
        studentsDict=defaultdict(int)
        for student in students:
            studentsDict[student]+=1
        for sandwiche in sandwiches:
            if studentsDict[sandwiche]>0:
                studentsDict[sandwiche]-=1
                remainedStudent -=1
            else:
                return remainedStudent
        return remainedStudent

        

File: 1802-number-of-students-unable-to-eat-lunch/number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        Queue<Integer> q = new LinkedList<>();
        for (int s : students) {
            q.add(s);
        }
        int eaten = 0;
        for (int s : sandwiches) {
            int att = 0;
            int len = q.size();
            while (att != len) {
                if (s == q.peek()) {
                    eaten++;
                    q.remove();
                    break;
                } else {
                    att++;
                    q.add(q.remove());

                }
            }
            if (len == att) {
                break;
            }

        }
        return sandwiches.length-eaten;

    }
}